gno.land/pkg/gnoweb/handler.go

```diff
@@ -264,6 +273,78 @@ func (h *WebHandler) GetRealmView(gnourl *weburl.GnoURL, indexData *components.I
 	})
 }
 
+// buildContributions returns the sorted list of contributions (packages and realms) for a user.
+func (h *WebHandler) buildContributions(username string) []components.UserContribution {
+	prefix := "@" + username
+	paths, err := h.Client.QueryPaths(prefix, 10000)
+	if err != nil {
+		h.Logger.Error("unable to query contributions", "user", username, "error", err)
+		return nil
+	}
+
+	contribs := make([]components.UserContribution, 0, len(paths))
+	for _, raw := range paths {
+		trimmed := strings.TrimPrefix(raw, h.Static.Domain)
+		u, err := weburl.Parse(trimmed)
+		if err != nil {
+			h.Logger.Error("bad contribution URL", "path", raw, "error", err)
+			continue
+		}
+		ctype := components.UserContributionTypePackage
+		if u.IsRealm() {
+			ctype = components.UserContributionTypeRealm
+		}
+		contribs = append(contribs, components.UserContribution{
+			Title: path.Base(raw),
+			URL:   raw,
+			Type:  ctype,
+			// TODO: size, description, date...
+		})
+	}
+
+	sort.Slice(contribs, func(i, j int) bool {
+		return contribs[i].Title < contribs[j].Title
+	})
+	return contribs
+}
+
+// GetUserView returns the user profile view for a given GnoURL.
+func (h *WebHandler) GetUserView(gnourl *weburl.GnoURL) (int, *components.View) {
+	username := strings.TrimPrefix(gnourl.Path, "/u/")
+	var content bytes.Buffer
+
+	// Render user profile realm
+	// TODO: render user profile realm instead of home realm
+	if _, err := h.Client.RenderRealm(&content, &weburl.GnoURL{Path: "/r/" + username + "/home"}, h.MarkdownRenderer); err != nil {
+		h.Logger.Debug("unable to render user realm", "error", err)
+	}
+
+	// Build contributions
+	contribs := h.buildContributions(username)
+
+	// Compute package counts
+	pkgCount := len(contribs)
+	realmCount := 0
+	for _, c := range contribs {
+		if c.Type.Id == components.UserContributionTypeRealm.Id {
+			realmCount++
+		}
+	}
+	pureCount := pkgCount - realmCount
```

Comment:
You could almost build `UserData` inside the `buildContributionData()` function to avoid iterating again for counting. Alternatively, you could move the content of function here or simply return a counter with the number of realms from `buildContribution() (realms int, contribs []contribs, err error)` or return two slices of pure/realm. This way, you can use len to count directly. 

This change won't significantly impact performance, so feel free to ignore this comment. However, I believe there should be a way to count this directly during the first iteration.